### PR TITLE
Add support of space-delimited list of events

### PR DESCRIPTION
**Pull request by rise2semi**
Add support of space-delimited list of events for triggerMethod.
The problem is described in #2937.

### CODE REVIEW

File: docs/marionette.functions.md [hunk 1]
```diff
@@ -173,6 +173,9 @@ All arguments that are passed to the triggerMethod call are passed along to both
 
 `triggerMethod("foo", bar)` will call `onFoo: function(bar){...})`
 
+To trigger several events and corresponding methods split the events by the space. Example:
+* `triggerMethod("render ready")` fires the "onRender" ans the "onReady" functions
```
Comment (paulfalgout) [inline docs/marionette.functions.md]:
```diff
- ans
+ and
```

Comment (paulfalgout) [inline docs/marionette.functions.md]: actually can we use `triggerMethod('foo bar')`?  Triggering render manually is probably not a great idea.

**Comment by rise2semi** [comment]
"render" event was chosen just as an example. I've changed the event names to more abstract.
Yes, we can use ```triggerMethod('foo bar')```. You can check it in this [jsfiddle](https://jsfiddle.net/b2Lrxqwr/1/) (I created a build with my changes).

**Comment by ahumphreys87** [comment]
@rise2semi this is great work! Could you please write a test for events bubbling up the view tree? to prove that childEvents can listen for events triggered like this.

**Comment by paulfalgout** [comment]
childViewEvents 😂

**Comment by paulfalgout** [comment]
Is there any perf concerns here? This is a pretty hot function

**Comment by JSteunou** [comment]
On the *map* / trigger multiple part I dont see any perf downside, at least no more than manually triggering multiple events at once from the view. Should not be *visible*.

My concern is more about the 1st test, if it cost a little, it's a little x all the time triggerMethod is called, and I think it is called a lot. How many does cost this line `if (eventSplitter.test(events)) {` ?

**Comment by rise2semi** [comment]
@paulfalgout I've added tests for childViewEvents.

I'm not sure that there will be performance issue with `eventSplitter.test(events)`, Backbone has the same logic for [events](https://github.com/jashkenas/backbone/blob/master/backbone.js#L148) and I haven't hear about performance issues in this place.

**Comment by scott-w** [comment]
Should we look to resolve these conflicts and bring this feature in for 3.1 or (potentially) 3.2?

File: src/trigger-method.js [hunk 2]
```diff
@@ -38,6 +35,31 @@ export function triggerMethod(event, ...args) {
...
+// `this.triggerMethod("foo:bar")` will trigger the "foo:bar" event and
+// call the "onFooBar" method.
+//
+// `this.triggerMethod("foo:bar baz:qux")` will trigger at first the "foo:bar"
+// event and call the "onFooBar" method and then trigger the "baz:qux" event
+// and call the "onBazQux" method
+export function triggerMethod(events, ...args) {
+  var result;
+
+  if (eventSplitter.test(events)) {
```
Comment (chchrist) [inline src/trigger-method.js]: do we need this if statement? ```"foo".split(/\s+/)``` returns an array

File: src/trigger-method.js [hunk 3]
```diff
@@ -38,6 +35,31 @@ export function triggerMethod(event, ...args) {
...
+// call the "onFoo" method.
+//
+// `this.triggerMethod("foo:bar")` will trigger the "foo:bar" event and
+// call the "onFooBar" method.
+//
+// `this.triggerMethod("foo:bar baz:qux")` will trigger at first the "foo:bar"
+// event and call the "onFooBar" method and then trigger the "baz:qux" event
+// and call the "onBazQux" method
+export function triggerMethod(events, ...args) {
+  var result;
```
Comment (chchrist) [inline src/trigger-method.js]: why not a const?

File: src/trigger-method.js [hunk 4]
```diff
@@ -7,20 +7,17 @@ import getOption from './utils/getOption';
 // split the event name on the ":"
 const splitter = /(^|:)(\w)/gi;
 
+// split the list of events on the space
+var eventSplitter = /\s+/;
```
Comment (chchrist) [inline src/trigger-method.js]: const?

**Comment by rafde** [comment]
@scott-w I am not a big fan of this implementation. triggerMethod is my 2nd perf woe and I am sure this implementation is going to create a hit in the perf. If I address this, it's pretty much wipes this code out.
I am leaning on closing this and starting over. @marionettejs/marionette-core any objections?

**Comment by scott-w** [comment]
Fair enough @rafde, I definitely see a lot of `triggerMethod` stuff happening when I run collection views inside collection views hammering the performance.

**Comment by rafde** [comment]
@rise2semi this branch is out of sync at the moment. If you are still interested in helping, please rebase and resolve conflicts.

**Comment by paulfalgout** [comment]
closing as it is stale

**Pull request closed without merging**